### PR TITLE
Stardew valley: Fix Queen of Sauce Cookbook conditions

### PR DESCRIPTION
**Pull request by agilbert1412**
## What is this fixing or adding?
Apparently the Queen of Sauce Cookbook can only spawn after collecting 100 golden walnuts. I wasn't aware of that the first time I wrote code for it.

This PR Extracts the walnut logic, in order to be able to use it as a content pack requirement, in order to be able to place this requirement on the Queen of Sauce Cookbook obtention.

The related locations have been removed from excluded GI worlds

## How was this tested?
Automated Tests

### CODE REVIEW

File: worlds/stardew_valley/logic/walnut_logic.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+class WalnutLogicMixin(BaseLogicMixin):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.walnut = WalnutLogic(*args, **kwargs)
+
+
+class CountLogicMixin:
+    pass
```
Comment (Jouramie) [inline worlds/stardew_valley/logic/walnut_logic.py:28]: Hmmm that's sus.

I think count is in fact in the `HasLogicMixin`.

**Review by Jouramie — COMMENTED** [review]
Except the weird class in walnut_logic.py, LGTM. I'll approve when it's sorted out.

I passed walnut_logic in a diff checker, and it's really just a copy-paste of logic.py, so all good on that part.

**Review by Jouramie — APPROVED** [review]

**Review by Exempt-Medic — APPROVED** [review]
Merged into `main` and did two generations (one with Ginger Island, one without). The changes LGTM, the refactor seems fine and the logic changes for QoS Cookbook match my understanding of the game and this code

**Review by black-sliver — APPROVED** [review]
Stuff just moved, so I am assuming it's fine and trust the peer reviews.

**Pull request merged**